metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -76,12 +77,15 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
     if(getSensorParserConfig() == null) {
       filter = new GenericMessageFilter();
     }
-    else if(filter == null) {
+    if(filter == null) {
       getSensorParserConfig().getParserConfig().putIfAbsent("stellarContext", stellarContext);
-      filter = Filters.get(getSensorParserConfig().getFilterClassName()
-              , getSensorParserConfig().getParserConfig()
-      );
+      if(!StringUtils.isEmpty(getSensorParserConfig().getFilterClassName())) {
```

Comment:
Well, it was confusing to see.  Why does GenericMessageFilter exist at all then?  At the very least I would think to put a comment in to explain why you are doing it that way for maintenance purposes if you leave it like that.